netsim/modules/_routing.py

```diff
@@ -261,35 +261,61 @@ def remove_vrf_interfaces(node: Box, proto: str) -> None:
-        n_data = topology.nodes[neighbor.node]
-        if proto in n_data.get('module',[]):                                # ... and check if at least one of them uses the IGP
-          node.vrfs[l.vrf][proto].active = True
-                                                                            # Cleanup IGP data
-  for vname,vdata in node.get('vrfs',{}).items():                           # ... iterate over the list of VRFs
+    if proto not in l:                                                # Not running the protocol on the interface?
+      continue
+    if 'vrf' not in l:                                                # Interface not in a VRF?
+      continue
+    if node.vrfs[l.vrf][proto] is True:                               # Handle 'force' the protocol by setting it to True
+      node.vrfs[l.vrf][proto] = { 'active': True }
+    elif node.vrfs[l.vrf][proto] is False:                            # Skip protocols disabled on VRF level
+      l.pop(proto,None)
+      continue
+    if not 'active' in node.vrfs[l.vrf][proto]:                       # Assume there are no IGP neighbors in this VRF
+      node.vrfs[l.vrf][proto].active = False
+    node.vrfs[l.vrf][proto] = node[proto] + node.vrfs[l.vrf][proto]   # Add node IGP parameters to VRF IGP parameters
+    data.append_to_list(
+      node.vrfs[l.vrf][proto],
+      list_name='interfaces',
+      item=data.get_box(l))                                           # Append a copy of the interface data
+    if vrf_unnumbered_check and \
+        l.get('ipv4',None) is True and \
+        l.get('_parent_vrf',None) != l.vrf:                           # Have to check the loopback source VRF?
+      data.append_to_list(unnum_err_list,l.vrf,l.ifname)              # ... remember we failed
+
+    l.pop(proto,None)                                                 # Finally, remove global IGP parameters from interface
+                                                                      # Next we need to find if the VRF instance of IGP matters
+    for neighbor in l.neighbors:                                      # ... iterate over the list of neighbors
+      n_data = topology.nodes[neighbor.node]
+      if proto in n_data.get('module',[]):                            # ... and check if at least one of them uses the IGP
+        node.vrfs[l.vrf][proto].active = True
+
+  # Time to cleanup IGP data
+  for vname,vdata in node.get('vrfs',{}).items():                     # ... iterate over the list of VRFs
     try:
-      proto_active = vdata.get(f'{proto}.active',False)                     # Get the IGP data for the VRF
-    except:                                                                 # ... assume 'not active' if get fails
+      proto_active = vdata.get(f'{proto}.active',False)               # Get the IGP data for the VRF
+    except:                                                           # ... assume 'not active' if get fails
```

Comment:
Does this ever happen? if this is to catch ```vdata``` being ```None```, then the code should be:
```
proto_active = vdata is not None and vdata.get(f'{proto}.active',False)
```

Reply:
It shouldn't (particularly now that we type-check everything), but do we want to change it and see what happens?

Reply:
> It shouldn't (particularly now that we type-check everything), but do we want to change it and see what happens?

I do - but it's easier for me to say that ;)

Reply:
> Does this ever happen? if this is to catch `vdata` being `None`, then the code should be:

The `vdata` is guaranteed to be set (at least in the **vrf** module), but `vdata[proto]` could be **False** (for example: `ospf: False`).

Rewrote the test to check for vdata[proto] type first and then use a get instead of a try/except.